common/djangoapps/student/models.py

```diff
@@ -11,18 +11,20 @@
 3. Add the migration file created in edx-platform/common/djangoapps/student/migrations/
```

Comment:
This is a core file. Let's remove the changes from this file if the impact of changes is not very optimized.

Reply:
@mabdullahz I have discussed impact of this change with @ahmed-faraz46 and we came to the conclusion that this check should be there in order to prevent extra event calls.

Reply:
@mabdullahz actually I do agree with you that we should avoid core changes, But what I see impact of these changes is significant and also there are already some core changes at the same place so this addition in core changes at same place is not gonna significantly impact on upgrade as well.